lib/storage/index_db.go

```diff
@@ -1518,7 +1518,7 @@ func (db *indexDB) searchMetricNameWithCache(dst []byte, metricID uint64) ([]byt
 		return dst, true
 	}
 
-	if db.s.shouldDeleteMissingMetricID(metricID) {
+	if db.s.wasMissingBefore(metricID) {
```

Comment:
WDYT of `wasMetricIDMissingBefore` ?

Reply:
See https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7036/commits/aa3fa935cbf2f0efbcbd0030e9da98aa0a89dec8